proof/crefine/ARM/SR_lemmas_C.thy

```diff
@@ -507,14 +507,21 @@ lemma fst_setCTE:
   shows   "P"
 proof -
   (* Unpack the existential and bind x, theorems in this.  Yuck *)
-  from fst_setCTE0 [where cte = cte, OF ct] guess s' by clarsimp
+  from fst_setCTE0 [where cte = cte, OF ct]
+  obtain s' where
+    "((), s')\<in>fst (setCTE dest cte s)"
+    "s' = s\<lparr>ksPSpace := ksPSpace s'\<rparr>"
+    "dom (ksPSpace s) = dom (ksPSpace s')"
+    "(\<forall>p \<in> dom (ksPSpace s').
+       case the (ksPSpace s p) of
+         KOTCB t \<Rightarrow> \<exists>t'. ksPSpace s' p = Some (KOTCB t') \<and> tcb_no_ctes_proj t = tcb_no_ctes_proj t'
+       | KOCTE _ \<Rightarrow> \<exists>cte. ksPSpace s' p = Some (KOCTE cte)
+       | _ \<Rightarrow> ksPSpace s' p = ksPSpace s p)"
+    by clarsimp
```

Comment:
what happened here? is the yuck comment still applicable?

Reply:
The `guess` command was removed. I think the `Yuck` is now more justified than before :-). (But really could also be removed, it doesn't add anything useful)

Reply:
Have removed the comment.